jupyterhub/orm.py

```diff
@@ -64,6 +64,8 @@ class Server(Base):
     """
     __tablename__ = 'servers'
     id = Column(Integer, primary_key=True)
+    
+    name = Column(Unicode(32), unique=True) # this is a unique reference to the server, probably a UUID4
```

Comment:
+1 on keeping db constraint inside the db.

Keep server_url unique (practically /user/{name}/server/{name}) I think would be enough.